src/proxy/offline/proxy.js

```diff
@@ -307,21 +307,17 @@ export default _.defaults(
                             if (data && data.length) {
                                 return self.sendSyncData(data);
                             }
-                        }).finally((data) => {
-                            self.syncInProgress = false;
```

Comment:
are you sure about this removal? the value is never reset once it has been set to true.
Maybe should you replace `finally` by `then` and move it after the `catch`

Reply:
Yes better adding it in `then` and `catch` to be safe

Reply:
put `self.syncInProgress = false;` both in then and catch